### PR TITLE
Add details to test_alert error dialogs.

### PR DESCRIPTION
**Pull request by timopollmeier**
An optional details field has been added to action_result dialogs which
 gets the text from the status_details element in error responses.
This is used in the response dialog for test_alert to show extra info
 when the alert script fails.

### CODE REVIEW

File: ng/src/gmp/http/transform/x2js.js [hunk 1]
```diff
@@ -72,6 +72,10 @@ class X2JsTransform extends Transform {
         }
 
         if (is_defined(envelope.action_result)) {
+          if (is_defined (envelope.action_result.details)
```
Comment (bjoernricks) [inline ng/src/gmp/http/transform/x2js.js]: Don't add spaces after functions and methods

File: ng/src/gmp/http/transform/x2js.js [hunk 2]
```diff
@@ -72,6 +72,10 @@ class X2JsTransform extends Transform {
...
+              && envelope.action_result.details !== '') {
```
Comment (bjoernricks) [inline ng/src/gmp/http/transform/x2js.js]: Can this ever happen? I don't like tests for empty string. Either it should not defined or it should contain useful information.

File: ng/src/web/components/notification/dialognotification.js [hunk 3]
```diff
@@ -112,6 +114,9 @@ class DialogNotification extends React.Component {
               />
               <ScrollableContent {...heightProps}>
                 {message}
+                {details
```
Comment (bjoernricks) [inline ng/src/web/components/notification/dialognotification.js]: Use shorter syntax `{details && <FootNote> ...}`

File: ng/src/gmp/http/transform/x2js.js [hunk 4]
```diff
@@ -72,6 +72,10 @@ class X2JsTransform extends Transform {
...
+            rej.setDetails (envelope.action_result.details);
```
Comment (bjoernricks) [inline ng/src/gmp/http/transform/x2js.js]: same as above don't add spaces before functions and methods.

File: ng/src/web/pages/alerts/component.js [hunk 5]
```diff
@@ -366,9 +366,25 @@ class AlertComponent extends React.Component {
...
+        if (is_defined(response) && is_defined(response.message)) {
+          onTestError(_('Testing the alert {{name}} failed: {{message}}',
+                        {
+                          name: alert.name,
+                          message: response.message,
+                          interpolation: { escapeValue: false }
+                        }),
+                      response.details,
+                     );
+        }
```
Comment (bjoernricks) [inline ng/src/web/pages/alerts/component.js]: Please update indentation
```javascript
onTestError(_('Testing the alert {{name}} failed: {{message}}', {
  name: alert.name,
  message: response.message,
  interpolation: {escapeValue: false},
}), response.details);
```

File: ng/src/web/pages/alerts/component.js [hunk 6]
```diff
@@ -366,9 +366,25 @@ class AlertComponent extends React.Component {
...
+        else {
+          onTestError(_('Testing the alert {{name}} failed.',
+                        {
+                          name: alert.name,
+                          interpolation: { escapeValue: false }
+                        }));
+        }
       }
```
Comment (bjoernricks) [inline ng/src/web/pages/alerts/component.js]: Please update indentation
```javascript
onTestError(_('Testing the alert {{name}} failed.', {
  name: alert.name,
  interpolation: {escapeValue: false},
}));
```

**Comment by bjoernricks** [comment]
I am really not sure about this change. First it would be nice if you split the changes for gsad and ng because both are really separated concerns. Second the Rejection class is very generic and it shouldn't get details if that is not applicable to all errors.

File: ng/src/web/pages/alerts/component.js [hunk 7]
```diff
@@ -366,9 +366,25 @@ class AlertComponent extends React.Component {
       if (is_defined(onTestSuccess)) {
         onTestSuccess(_('Testing the alert {{name}} was successful.', alert));
       }
-    }, () => {
+    }, response => {
```
Comment (bjoernricks) [inline ng/src/web/pages/alerts/component.js]: please use rejection as variable name here. that's clearer which object you will get.

**Comment by bjoernricks** [comment]
Instead of adding a generic details you could just adjust the alert test method:

```javascript
test({id}) {
  return this.httpPost({
    cmd: 'test_alert',
    id,
  })
  .catch(rejection => {
    Promise.reject({
      message: rejection.message,
      details: rejection.action_result.details,
    });
  });
}
```

**Comment by timopollmeier** [comment]
Since it's better to split off the gsad part and the ng part needs to be revised, I'm closing this pull request. I've already created #296 for the gsad part and re-implement the ng part for the dialog.

**Pull request closed without merging**